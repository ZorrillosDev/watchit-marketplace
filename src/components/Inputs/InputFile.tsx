// MUI IMPORTS
import {
  Box, Button, InputLabel, styled, ButtonProps,
  InputLabelProps, Typography, BoxProps, TypographyProps
} from '@mui/material'

/* eslint-disable  @typescript-eslint/strict-boolean-expressions */

// ===========================|| INPUT - FILE ||=========================== //

export const InputFileElement = styled('input')({
  display: 'none'
})

export const InputFileIcon = styled(Box)<BoxProps>(({ theme }) => ({
  svg: {
    color: theme.palette.text.secondary,
    width: '2rem',
    height: '2rem'
  }
}))

export const InputFileLabelText = styled(Typography)<TypographyProps>(({ theme }) => ({
  position: 'absolute',
  left: 0,
  top: '-2rem'
}))

export const InputFileLabel = styled(InputLabel, {
  shouldForwardProp: (prop) => prop !== 'isFull'
})<InputLabelProps & { filled: boolean, isFull: boolean }>(({ theme, filled, isFull }) => ({
  border: `2px dashed ${theme.palette.divider}`,
  minHeight: '150px',
  width: '100%',
  padding: filled ? '16px' : '32px 60px',
  position: 'relative',
  borderRadius: '16px',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  flexDirection: 'column',
  overflow: 'visible',
  marginBottom: '1.5rem',
  marginTop: '1.5rem',
  cursor: 'pointer',
  '&:hover': {
    'h5, svg': {
      color: theme.palette.text.primary
    },
    borderColor: theme.palette.text.primary
  },
  '&.Mui-error': {
    'h5, svg': {
      color: theme.palette.error.main
    },
    borderColor: theme.palette.error.main
  },
  '& > .MuiPaper-root': {
    maxWidth: isFull ? '100%' : '16rem'
  }
}))

export const InputFileButton = styled(Button)<ButtonProps>(() => ({
  marginBottom: '-1.5rem',
  transform: 'translateY(2rem)',
  width: '14rem',
  height: '3rem',
  borderRadius: '12px !important',
  boxShadow: '0 5px 9px rgb(0 0 0 / 20%)',
  fontWeight: 600
}))
