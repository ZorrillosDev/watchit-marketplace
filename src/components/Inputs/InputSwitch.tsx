// REACT IMPORTS
import React, { FC, useState } from 'react';

// MUI IMPORTS
import {
  Accordion, AccordionDetails, AccordionSummary, SwitchProps,
  Box, styled, Typography, BoxProps, Switch,
} from '@mui/material';

/* eslint-disable  @typescript-eslint/strict-boolean-expressions */

// ===========================|| INPUTS - SWITCH ||=========================== //

export interface InputSwitchProps {
  label: string | JSX.Element
  subtitle: string | JSX.Element
  content?: JSX.Element
  defaultChecked?: boolean
}

export const InputSwitch: FC<InputSwitchProps> = (props): JSX.Element => {
  const [checked, setChecked] = useState(props.defaultChecked ?? false);

  const handleChange = (event: any): void => {
    setChecked(event.target.checked);
  };

  return (
    <InputSwitchWrapper>
      <Box display='flex' flexDirection='row' alignItems='center' width={1}>
        <Box display='flex' flexDirection='column' flexGrow={1}>
          <Typography variant='body2' color='primary' fontWeight={600}>{props.label}</Typography>
          <Typography variant='body1' color='primary' sx={{ opacity: 0.8 }}>{props.subtitle}</Typography>
        </Box>
        <InputSwitchElement
          checked={checked}
          onChange={handleChange}
          inputProps={{ 'aria-label': 'controlled' }}
        />
      </Box>
      <Accordion expanded={checked} sx={{ m: '0 !important', width: 1, '&:before': { display: 'none' } }}>
        <AccordionSummary sx={{ minHeight: '0px !important', height: 0 }} />
        <AccordionDetails sx={{ p: 0, pt: props.content ? '1rem' : 0 }}>
          {props.content}
        </AccordionDetails>
      </Accordion>
    </InputSwitchWrapper>
  );
};

const InputSwitchWrapper = styled(Box)<BoxProps>(() => ({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  flexDirection: 'column',
  width: '100%',
}));

const InputSwitchElement = styled(Switch)<SwitchProps>(({ theme }) => ({
  width: 42,
  height: 26,
  padding: 0,
  '& .MuiSwitch-switchBase': {
    padding: 0,
    margin: 2,
    transitionDuration: '300ms',
    '&.Mui-checked': {
      transform: 'translateX(16px)',
      color: '#fff',
      '& + .MuiSwitch-track': {
        backgroundColor: theme.palette.primary.main,
        opacity: 1,
        border: 0,
      },
      '&.Mui-disabled + .MuiSwitch-track': {
        opacity: 0.5,
      },
    },
    '&.Mui-focusVisible .MuiSwitch-thumb': {
      color: theme.palette.primary.main,
      border: '6px solid #fff',
    },
    '&.Mui-disabled .MuiSwitch-thumb': {
      color:
        theme.palette.mode === 'light'
          ? theme.palette.grey[100]
          : theme.palette.grey[600],
    },
    '&.Mui-disabled + .MuiSwitch-track': {
      opacity: theme.palette.mode === 'light' ? 0.7 : 0.3,
    },
  },
  '& .MuiSwitch-thumb': {
    boxSizing: 'border-box',
    width: 22,
    height: 22,
  },
  '& .MuiSwitch-track': {
    borderRadius: 26 / 2,
    backgroundColor: theme.palette.mode === 'light' ? '#E9E9EA' : '#39393D',
    opacity: 1,
    transition: theme.transitions.create(['background-color'], {
      duration: 500,
    }),
  },
}));
