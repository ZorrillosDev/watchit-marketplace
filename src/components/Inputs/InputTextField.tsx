// MUI IMPORTS
import { TextField, styled, TextFieldProps } from '@mui/material'

// ===========================|| INPUT - TEXT - FIELD ||=========================== //

export const InputTextField = styled(TextField)<TextFieldProps>(({ theme }) => ({
  width: '100%',
  'label, input, textarea': {
    color: theme.palette.text.primary,
    fontWeight: 600
  }
}))
