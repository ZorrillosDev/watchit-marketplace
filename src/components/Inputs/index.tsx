// ===========================|| POSTER EXPORTS ||=========================== //

export { InputTextField as default } from '@components/Inputs/InputTextField'
export { InputSwitch } from '@components/Inputs/InputSwitch'
export * from '@components/Inputs/InputFile'
