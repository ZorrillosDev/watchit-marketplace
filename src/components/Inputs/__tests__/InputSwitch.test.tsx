import React from 'react'
import { shallow } from 'enzyme'
import { InputSwitch } from '@components/Inputs'
import { Typography } from '@mui/material'

/* eslint-disable  @typescript-eslint/consistent-type-assertions */

describe('<InputSwitch />', () => {
  it('should render', () => {
    const component = shallow(<InputSwitch title='hello world' subtitle='test hehe' />)

    expect(component).toMatchSnapshot()
  })

  it('should have movie create form switch with valid title and subtitle text', () => {
    const title = 'hello world'
    const subTitle = 'test hehe'
    const component = shallow(<InputSwitch title={title} subtitle={subTitle} />)

    expect(component.find(Typography).at(0).text()).toMatch(title)
    expect(component.find(Typography).at(1).text()).toMatch(subTitle)
  })
})
