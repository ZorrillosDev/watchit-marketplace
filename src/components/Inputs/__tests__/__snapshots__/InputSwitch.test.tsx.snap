// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputSwitch /> should render 1`] = `ShallowWrapper {}`;
